File: зад2.java
import java.util.Scanner;
public class Main{
  public static void main(String[] args) {
    Scanner scanner = new Scanner(System.in);
    System.out.print("Введите строку:  ");
    String str = scanner.nextLine();
    char[] rings = str.toCharArray();
	//Проверка
	for (int i = 0; i < rings.length ; i += 2) {
		if (rings[i]=='R' || rings[i]=='B' || rings[i]=='W') {
			if (rings[i + 1] < '0' || rings[i + 1]>'9') {
				System.out.print("Значение было введено неверно");
			}
		}else {
			System.out.print("Значение было введено неверно");
		}
	}
	
int[] sticks = {1, 1, 1, 1, 1, 1, 1, 1, 1, 1};
	//Считаем кол-во колец на каждом стержне
for (int i = 0; i < rings.length; i += 2) {
	
		if (rings[i] == 'R') {
			sticks[rings[i + 1] - '0'] *= 2;
		}
		else if (rings[i] == 'W') {
			sticks[rings[i + 1] - '0'] *= 3;
		}
		else if (rings[i] == 'B') {
			sticks[rings[i + 1] - '0'] *= 5;
		}
	}
	int counter = 0;
	//На скольких кольцах стержни одного цвета
	for (int i=0; i<sticks.length;i++) {
		if (sticks[i] % 30 == 0) {
			counter++;
		}
	}
	System.out.print(counter);
  }
}

File: зад.2.kt
/*	for (i in 0..rings.count() step 2) {
		if (rings[i].equals("R") || rings[i].equals("R") || rings[i].equals("W")) {
			if (rings[i + 1] < "0" || rings[i + 1]>"9") {
				print("error")
			}
		} else {
			print("error")
		}
	}
*/	
import kotlin.math.sqrt 
import java.util.Scanner 
fun main()  {
    val scanner = Scanner(System.`in`)
    print("enter string: ") 
    var rings = ""
    rings = scanner.nextLine()              
    scanner.close(); 

val sticks: Array<Int> = arrayOf(1, 1, 1, 1, 1, 1, 1, 1, 1, 1)
for (i in 0..rings.count()-2 step 2) {
		if (rings[i] == 'R') {
			sticks[rings[i + 1].toInt()-48] *= 2
		} else if (rings[i] == 'W') {
			sticks[rings[i + 1].toInt()-48] *= 3
		} else if (rings[i] == 'B'){
			sticks[rings[i + 1].toInt()-48] *= 5
		}
	}
	var counter = 0
	for (i in 0..sticks.size-1) {
		if (sticks[i] % 30 == 0) {
			counter++
			
		}
	}
	print(counter)
 }
